packages/core/src/api/oauth2.ts

```diff
@@ -121,4 +124,24 @@ export class OAuth2API {
 			signal,
 		}) as Promise<RESTGetAPIOAuth2CurrentAuthorizationResult>;
 	}
+
+	/**
+	 * Revokes an OAuth2 token
+	 * 
+	 * @see {@link https://discord.com/developers/docs/topics/oauth2#authorization-code-grant-token-revocation-example}
+	 * @param body - The body of the token revocation request
+	 * @param options - The options for the token revocation request
+	 */
+	public async revokeToken(applicationId: Snowflake, applicationSecret: string, body: RESTPostOAuth2TokenRevocation, { signal }: Pick<RequestData, 'signal'> = {}) {
+	 return this.rest.post(Routes.oauth2TokenRevocation(), {
+	  body: makeURLSearchParams(body),
+	  passThroughBody: true,
+	  headers: {
+	   'Content-Type': 'application/x-www-form-urlencoded',
+	   'Authorization': `Basic ${Buffer.from(`${applicationId}:${applicationSecret}`).toString('base64')}`,
```

Comment:
We can use the `btoa()` global instead of the node-specific `Buffer.from()`
```suggestion
	   'Authorization': `Basic ${btoa(`${applicationId}:${applicationSecret}`)}`,
```